x-pack/plugins/cloud_security_posture/server/saved_objects/data_views.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  ElasticsearchClient,
+  ISavedObjectsRepository,
+  SavedObject,
+  type KibanaRequest,
+  type Logger,
+} from '@kbn/core/server';
+import { DataViewAttributes } from '@kbn/data-views-plugin/common';
+import { SpacesServiceStart } from '@kbn/spaces-plugin/server';
+import { DataViewsServerPluginStart } from '@kbn/data-views-plugin/server';
+
+import {
+  CDR_MISCONFIGURATIONS_DATA_VIEW_ID_PREFIX,
+  CDR_MISCONFIGURATIONS_DATA_VIEW_NAME,
+  CDR_MISCONFIGURATIONS_INDEX_PATTERN,
+  CDR_VULNERABILITIES_DATA_VIEW_ID_PREFIX,
+  CDR_VULNERABILITIES_DATA_VIEW_NAME,
+  CDR_VULNERABILITIES_INDEX_PATTERN,
+} from '../../common/constants';
+
+const PREVIOUS_VERSION_INDEX_PATTERNS = {
+  [CDR_MISCONFIGURATIONS_DATA_VIEW_ID_PREFIX]: [],
+  [CDR_VULNERABILITIES_DATA_VIEW_ID_PREFIX]: [],
+};
+const CURRENT_VERSION_INDEX_PATTERNS: Record<string, string> = {
+  [CDR_MISCONFIGURATIONS_DATA_VIEW_ID_PREFIX]: CDR_MISCONFIGURATIONS_INDEX_PATTERN,
+  [CDR_VULNERABILITIES_DATA_VIEW_ID_PREFIX]: CDR_VULNERABILITIES_INDEX_PATTERN,
+};
+
+const DATA_VIEW_TIME_FIELD = '@timestamp';
+
+const getDataViewSafe = async (
+  soClient: ISavedObjectsRepository,
+  currentSpaceId: string,
+  currentSpaceDataViewId: string,
+  indexPattern: string,
+  logger: Logger
+): Promise<SavedObject<DataViewAttributes> | undefined> => {
+  try {
+    const dataView = await soClient.get<DataViewAttributes>(
+      'index-pattern',
+      currentSpaceDataViewId,
+      {
+        namespace: currentSpaceId,
+      }
+    );
+
+    // Verify user didn't edit the index pattern
+    const currentIndexPatterns = dataView.attributes.title.split(',');
+    const requiredIndexPatterns = indexPattern.split(',');
+
+    requiredIndexPatterns.every((element) => {
```

Comment:
what about moving it out from `getDataViewSafe` to a separate function? to not mix getting the data view with the migration? In the end, the function returns the data view prior to the migration and anyway, it's better not to mix two different operations together in my opinion

Reply:
This migration will be removed based on [this discussion](https://github.com/elastic/kibana/pull/188625/files#r1691492680).